### PR TITLE
Added unit tests 

### PR DESCRIPTION
**Pull request by ysnarafat**
Add unit tests for issue #8

### CODE REVIEW

File: InvoiceProcessor.Application/Queries/Outbox/GetPendingProcess/QueryHandler.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public async Task<List<Model>> Handle(Query request, CancellationToken cancellat
                 var pendingItems = await _outboxStorage.GetPendingItemsAsync(cancellationToken);
 
                 return pendingItems?.Select(x => new Model(
-                    guid: x.Guid.Value,
+                    guid: x.Guid.GetValueOrDefault(),
```
Comment (abuzaforfagun) [inline InvoiceProcessor.Application/Queries/Outbox/GetPendingProcess/QueryHandler.cs]: What's the reason behind this modification?

File: InvoiceProcessor.Tests/Application/Queries/Outbox/GetPendingProcessTests.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        [Fact]
+        public async Task GetPendingProcess_Should_Return_Empty_When_Pending_Process_Unavailable()
+        {
+            // Arrange
+            List<OutboxItem> outboxItems = null;
+            List<Model> models = null;
+            var handler = new QueryHandler(_outboxStorageMock.Object);
+            _outboxStorageMock.Setup(x => x.GetPendingItemsAsync(new CancellationToken()))
```
Comment (abuzaforfagun) [inline InvoiceProcessor.Tests/Application/Queries/Outbox/GetPendingProcessTests.cs:31]: We need to do the setup the mock before using the mock.

File: InvoiceProcessor.Tests/Application/Queries/Outbox/GetPendingProcessTests.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        {
+            _outboxStorageMock = new Mock<IOutboxStorage>();
+        }
+
+        [Fact]
+        public async Task GetPendingProcess_Should_Return_Empty_When_Pending_Process_Unavailable()
+        {
+            // Arrange
+            List<OutboxItem> outboxItems = null;
+            List<Model> models = null;
```
Comment (abuzaforfagun) [inline InvoiceProcessor.Tests/Application/Queries/Outbox/GetPendingProcessTests.cs]: When there is no pending invoice, it returns an empty List not null. And when you want to pass null, you directly can pass null insisted of creating another variable.

File: InvoiceProcessor.Tests/Application/Queries/Outbox/GetPendingProcessTests.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+            var handler = new QueryHandler(_outboxStorageMock.Object);
+            _outboxStorageMock.Setup(x => x.GetPendingItemsAsync(new CancellationToken()))
+                .ReturnsAsync(outboxItems);
+
+            // Act
+            var results = await handler.Handle(new Query(), new CancellationToken());
+
+            // Assert
+            Assert.NotStrictEqual(models, results);
```
Comment (abuzaforfagun) [inline InvoiceProcessor.Tests/Application/Queries/Outbox/GetPendingProcessTests.cs]: I think the assertion is not working correctly, when you remove one item from models variable, it still pass.

File: InvoiceProcessor.Tests/Application/Queries/Outbox/GetProcessStatusTests.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        [Theory]
+        [MemberData(nameof(Data))]
+        public async Task GetProcessStatus_Should_Return_Process_Status(OutBoxStatus status)
+        {
+            // Arrange
+            var message = Guid.Empty;
+            var handler = new QueryHandler(_processStatusRepositoryMock.Object);
+            _processStatusRepositoryMock.Setup(x => x.GetProcessStatus(message))
```
Comment (abuzaforfagun) [inline InvoiceProcessor.Tests/Application/Queries/Outbox/GetProcessStatusTests.cs]: When setting up mock for generic values, it's better to use It.IsAny<Type>()

File: InvoiceProcessor.Tests/Infrastructure/Outbox/Services/OutBoxServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            // Assign 
+            var guid = Guid.NewGuid();
+            var payload = new OutBoxEvent(guid);
+            var model = new OutBoxModel(payload, OutBoxStatus.Pending, "abcd");
+
+            // Act
+            await _outBoxServiceMock.Upsert(model);
+
+            // Assert
+            _storageMock.Verify();
```
Comment (abuzaforfagun) [inline InvoiceProcessor.Tests/Infrastructure/Outbox/Services/OutBoxServiceTests.cs]: What it would verify?

**Review by abuzaforfagun — CHANGES_REQUESTED** [review]

File: InvoiceProcessor.Application/Queries/Outbox/GetPendingProcess/QueryHandler.cs (hunk 1 above)
Comment (ysnarafat) [reply InvoiceProcessor.Application/Queries/Outbox/GetPendingProcess/QueryHandler.cs]: Previously OutboxItem Guid property was nullable. At that time it was throwing exception while accessing nullable value.

**Review by abuzaforfagun — APPROVED** [review]

**Pull request merged**